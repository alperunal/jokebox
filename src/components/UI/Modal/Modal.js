import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Close from '../../../assets/icons/close.png';

const Modal = props => {
  const { children, closeModal } = props;
  return (
    <StyledModal>
      <Window>
        <Actions>
          <CloseIcon onClick={closeModal} src={Close} />
        </Actions>
        <Content>{children}</Content>
      </Window>
    </StyledModal>
  );
};

const StyledModal = styled.div`
  background-color: rgba(0, 0, 0, 0.6);
  z-index: 1000;
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
`;
const Window = styled.div`
  width: 900px;
  height: 400px;
  background-color: #ffffff;
  padding: 20px;
`;

const Actions = styled.span`
  display: flex;
  flex-direction: row-reverse;
`;

const CloseIcon = styled.img`
  width: 16px;
  height: 16px;
  cursor: pointer;
`;

const Content = styled.div``;

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  closeModal: PropTypes.func.isRequired,
};

export default Modal;
