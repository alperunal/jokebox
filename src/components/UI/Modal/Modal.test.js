import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

it('renders Modal correctly', () => {
  const func = () => {};
  const wrapper = shallow(<Modal closeModal={func}>TEST</Modal>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains('TEST')).toBe(true);
});
