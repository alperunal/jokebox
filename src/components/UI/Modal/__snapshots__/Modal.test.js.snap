// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      <styled.img
        onClick={[Function]}
        src="close.png"
      />
    </styled.span>
    <styled.div>
      TEST
    </styled.div>
  </styled.div>
</styled.div>
`;
